src/Plugin/Patches.php

```diff
@@ -442,8 +442,11 @@ protected function getAndApplyPatch(HttpDownloader $downloader, $install_path, $
             // --no-backup-if-mismatch here is a hack that fixes some
             // differences between how patch works on windows and unix.
             $patch_options = '--no-backup-if-mismatch';
-            if (PHP_OS_FAMILY == 'BSD') {
-                $patch_options = '--posix --batch';
+            // sniff if we're using GNU patch
+            $patchVersion = '';
+            $this->executor->execute('patch --version', $patchVersion);
+            if (strpos($patchVersion, 'GNU patch') === false) {
```

Comment:
This condition should be `!==` because these options only work on GNU

Reply:
FreeBSD has a non-GNU patch (version is `patch 2.0-12u11 FreeBSD`) and definitely has both the --posix and --batch options: https://man.freebsd.org/cgi/man.cgi?patch(1)

Reply:
@joelpittet  BSD patch in FreeBSD has these options too.  I don't know why wouldn't have at least pulled the latest version of patch from FreeBSD.  Does the Ventura patch have --posix or --no-backup-if-mismatch?  Are you able to send me a copy of the2.0-12u11-Apple patch man page and the output of the patch usage.  I can replace --batch with '-t', but need a replacement for --posix or --no-backup-if-mismatch for the Apple patch case.

Reply:
@arcanumbridge On Ventura 13.2 it appears to have all the flags, so maybe not straightforward to "sniff"
```
$ patch --version
patch 2.0-12u11-Apple
```

```
$ man patch
NAME
     patch – apply a diff file to an original

SYNOPSIS
     patch [-bCcEeflNnRstuv] [-B backup-prefix] [-D symbol] [-d directory]
           [-g vcs-option] [-F max-fuzz] [-i patchfile] [-o out-file]
           [-p strip-count] [-r rej-name] [-T | -Z] [-V t | nil | never | none]
           [-x number] [-Y prefix] [-z backup-ext] [--quoting-style style] [--posix]
           [origfile [patchfile]]
     patch <patchfile
...
     --no-backup-if-mismatch
             This option negates --backup-if-mismatch, creating backups for every
             file unless backups have been disabled with --posix or -V none.
     --posix
             Enables strict IEEE Std 1003.1-2008 (“POSIX.1”) conformance,
             specifically:

             1.   Backup files are not created unless the -b option is specified.

             2.   If unspecified, the file name used is the first of the old, new
                  and index files that exists.
     -t, --batch
             Similar to -f, in that it suppresses questions, but makes some
             different assumptions: skip patches for which a file to patch cannot be
             found (the same as -f); skip patches for which the file has the wrong
             version for the "Prereq": line in the patch; and assume that patches
             are reversed if they look like they are.
```

Reply:
Ah this looks like potentially good news, if at some point Apple updated their patch to the same version as FreeBSD?